packages/kbn-data-service/src/search/tabify/tabify_docs.ts

```diff
@@ -147,28 +151,73 @@ export function flattenHit(hit: Hit, indexPattern?: DataView, params?: TabifyDoc
 
   // Use a proxy to make sure that keys are always returned in a specific order,
   // so we have a guarantee on the flattened order of keys.
-  return makeProxy(flat, indexPattern);
+  return makeProxy(flat, indexPattern, params?.flattenedFieldsComparator);
 }
 
-function makeProxy(flat: Record<string, any>, indexPattern?: DataView) {
-  function comparator(a: string | symbol, b: string | symbol) {
-    const aIsMeta = indexPattern?.metaFields?.includes(String(a));
-    const bIsMeta = indexPattern?.metaFields?.includes(String(b));
+export const getFlattenedFieldsComparator = (indexPattern?: DataView) => {
```

Comment:
This approach of having a comparator that's instantiated outside and passed down to `flattenHit` is convoluted, but it brings the total time spent sorting from ~1700 ms to ~600 ms in my tests, while preserving the case insensitive custom sorting logic.

I did try some other approaches, but this was the most performant one. For reference and in case we want to consider simpler alternatives, these were the approximate timings:
- Instantiate a static `Intl.Collator` outside the method and share it for `localeCompare` calls: ~750 ms
- Call `toLowerCase` on every sort vs caching: ~750ms
- Remove case insensitive sorting and compare the strings directly: ~400ms

Reply:
Have you tried with `toLocaleLowerCase` that could bring in the benefit of the `localeCompare` to your cached method?

Reply:
I didn't, but I just gave it a try now. It seems like the performance impact is small vs `toLowerCase`, maybe ~50-75 ms in my tests, but I'm not sure we need locale aware comparisons for sorting field names. I believe `localeCompare` was originally only used to support the case insensitive sorting need.